lib/docker-watcher.js

```diff
@@ -37,13 +37,12 @@ DockerWatcher.prototype._onContainersListReceived = function (list) {
   var diff = false
 
   list.forEach(function (cnt, id) {
-    diff = !this.containers[id] || (_.isEqual(list[id].getComparableInfos(), this.containers[id].getComparableInfos()) === false)
+    diff = diff || !this.containers[id] || (_.isEqual(list[id].getComparableInfos(), this.containers[id].getComparableInfos()) === false)
```

Comment:
this was resetting diff to false if the last container didn't have a diff, even if there was a diff in one of the containers. this made it so if a container was restarted with a different port assigned, it would not get picked up.